lib/statsig/api_client.ex

```
@@ -1,55 +1,105 @@
+            {:error, :unexpected_error, error}
+
+          {:error, error} ->
+            Logger.error("Unexpected error: #{inspect(error)}")
+            {:error, :unexpected_error, error}
+        end
+
+      {:error, reason} ->
+        Logger.error("Failed to get API key: #{reason}")
+        {:error, :missing_api_key, reason}
+    end
+  end
+
+  def push_logs(logs) do
+    case api_key() do
+      {:ok, key} = key_result ->
+        base_url = api_url(@default_logging_api_url)
+
+        url =
+          URI.new!(base_url)
+          |> URI.append_path("/rgstr")
+          |> URI.to_string()
+
+        case Req.post(url: url, json: %{events: logs}, headers: headers(key)) do
+          {:ok, %{status: code}} when code < 300 ->
+            {:ok, []}
+
+          {:ok, response} ->
+            Logger.error(
+              "Failed to push logs: status #{response.status}, body: #{inspect(response.body)}"
+            )
+
+            {:error, logs}
+
+          {:error, error} ->
+            Logger.error("Failed to push logs: #{inspect(error)}")
+            {:error, logs}
+        end
+
+      {:error, reason} ->
+        Logger.error("Failed to get API key: #{reason}")
+        {:error, logs}
     end
   end
 
-  def push_logs(api_key, logs) do
-    base_url = get_api_url(@default_logging_api_url)
-    url = "#{base_url}rgstr"
-    result = Req.post(
-      url: url,
-      json: %{"events" => logs},
-      headers: headers(api_key)
-    ) |> case do
-      {:ok, %{status: code}} when code < 300 -> {:ok, []}
-      _ -> {:error, logs}
+  defp api_key() do
+    case Application.get_env(:statsig, :api_key) do
+      nil ->
+        Logger.error("Statsig API key is not configured")
+        {:error, "API key not configured"}
+
+      key when is_binary(key) ->
+        {:ok, key}
+
+      key ->
+        Logger.error("Invalid Statsig API key format: #{inspect(key)}")
+        {:error, "Invalid API key format"}
```

same comment about error reasons. 

Question, if the key isn't set, is this error recoverable? You might just want to raise an exception here.

---

good point